src/masternodes/anchors.cpp

```diff
@@ -780,6 +780,15 @@ bool CAnchorIndex::DbErase(uint256 const & hash)
     return db->Erase(std::make_pair(DB_ANCHORS, hash));
 }
 
+const CAnchorIndex::AnchorRec* CAnchorIndex::GetLatestAnchorUpToDeFiHeight(THeight blockHeightDeFi) const
+{
+    AssertLockHeld(cs_main);
+
+    auto & index = anchors.get<AnchorRec::ByDeFiHeight>();
+    auto it = index.lower_bound(blockHeightDeFi);
+    return (it != index.begin()) ? &(*(--it)) : nullptr;
```

Comment:
I think here should compare `it` with `index.end`.

template<typename CompatibleKey>
iterator lower_bound(const CompatibleKey& x)const;
Requires: CompatibleKey is a compatible key of key_compare.
Effects: Returns an iterator pointing to the first element with key not less than x, or end() if such an element does not exist.
Complexity: O(log(n)).